### PR TITLE
new package: py-python-ldap

### PR DESCRIPTION
**Pull request by Sinan81**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-python-ldap/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    homepage = "https://www.python-ldap.org/en/python-ldap-3.2.0/"
+    url      = "https://pypi.io/packages/source/p/python-ldap/python-ldap-3.2.0.tar.gz"
+
+    version('3.2.0', sha256='7d1c4b15375a533564aad3d3deade789221e450052b21ebb9720fb822eccdb8e')
+    version('3.0.0', sha256='86746b912a2cd37a54b06c694f021b0c8556d4caeab75ef50435ada152e2fbe1')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
+    depends_on('py-pyasn1', type='run')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-python-ldap/package.py]:
```suggestion
    depends_on('py-pyasn1@0.3.7:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-python-ldap/package.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    depends_on('py-pyasn1-modules', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-python-ldap/package.py]:
```suggestion
    depends_on('py-pyasn1-modules@0.1.5:', type=('build', 'run'))
```

**Comment by Sinan81** [comment]
I will be creating openldap spack package in the following days. Based on the documentation, it seems python-ldap requires openldap client type installation.

**Comment by Sinan81** [comment]
I think this PR is ready to merge.

**Comment by Sinan81** [comment]
Pinging @adamjstewart

File: var/spack/repos/builtin/packages/py-python-ldap/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    servers from Python programs.
+    """
+
+    homepage = "https://www.python-ldap.org/en/python-ldap-3.2.0/"
+    url      = "https://pypi.io/packages/source/p/python-ldap/python-ldap-3.2.0.tar.gz"
+
+    version('3.2.0', sha256='7d1c4b15375a533564aad3d3deade789221e450052b21ebb9720fb822eccdb8e')
+    version('3.0.0', sha256='86746b912a2cd37a54b06c694f021b0c8556d4caeab75ef50435ada152e2fbe1')
+
+    depends_on('openldap+client_only', type=('build', 'link', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-python-ldap/package.py]: Can it not be built with `openldap~client_only`?

Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-python-ldap/package.py]: Not sure, but I think the intention of this package is to be able to manage or interact with an openldap installation on a server. Hence, client_only installation of openldap as suggested by developers should be sufficient. If you think that it's a good idea, I can create a client_only variant (enabled by default) triggering a client_only building of openldap.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-python-ldap/package.py]: Nah, that's okay. I'll defer to the developers if they recommend the client-only build.

**Pull request merged**